mmcv/ops/csrc/common/cuda/knn_cuda_kernel.cuh

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) OpenMMLab. All rights reserved
+// Modified from
+// https://github.com/CVMI-Lab/PAConv/tree/main/scene_seg/lib/pointops/src/knnquery_heap
+#ifndef KNN_CUDA_KERNEL_CUH
+#define KNN_CUDA_KERNEL_CUH
+
+#ifdef MMCV_USE_PARROTS
+#include "parrots_cuda_helper.hpp"
+#else
+#include "pytorch_cuda_helper.hpp"
+#endif
+
+#define DIVUP(m, n) ((m) / (n) + ((m) % (n) > 0))
+
+__device__ void swap_float(float *x, float *y) {
```

Comment:
I mean you can define a swap template for both float and int.
And, since this swap is frequently used in `reheap` and `sort`, you can mark it as an inline function.